polyfill/lib/calendar.mjs

```diff
@@ -1013,22 +995,30 @@ const nonIsoHelperBase = {
   // All built-in calendars except Chinese/Dangi and Hebrew use an era
   hasEra: true,
   monthDayFromFields(fields, overflow, cache) {
-    let { year, month, monthCode, day, era, eraYear } = fields;
+    let { monthCode, day } = fields;
     if (monthCode === undefined) {
+      let { year, era, eraYear } = fields;
       if (year === undefined && (era === undefined || eraYear === undefined)) {
         throw new TypeError('`monthCode`, `year`, or `era` and `eraYear` is required');
       }
-      ({ monthCode, year } = this.adjustCalendarDate({ year, month, monthCode, day, era, eraYear }, cache, overflow));
+      // Apply overflow behaviour to year/month/day, to get correct monthCode/day
+      ({ monthCode, day } = this.isoToCalendarDate(this.calendarToIsoDate(fields, overflow, cache), cache));
```

Comment:
It's been a while since I wrote this so I may be wrong, but if I remember correctly `this.adjustCalendarDate` should fill in `monthCode`. Is the problem that `day` isn't constrained by `this.adjustCalendarDate` for the Chinese calendar?

Reply:
Not sure about the Chinese calendar for this one, but the previous code only using `adjustCalendarDate` tripped a failure in the Hebrew calendar tests I wrote. It didn't take the Hebrew calendar's short months into account, only clamping the `day` to `maximumMonthLength`. I didn't see an obvious way to get it to handle that besides duplicating the whole `isoToCalendarDate(calendarToIsoDate(...))` logic, but I'm open to suggestions. (Although I might rather implement those suggestions in a followup PR)